### DISCUSSION TITLE
Should other methods such as `addRows` trigger the `valuesUpdated` event?

### DISCUSSION THREAD

**Discussion by MartinDawson** [handsontable/hyperformula#846] [API Questions]
Currently `valuesUpdated` event only triggers on `setCellContents`.

It does not trigger on other methods that cause formulas to be moved around and updated.

Technically when `addRows` is called it does updated the hyperformula values and this event should be called imo.

If not I am thinking maybe other events for more methods such as a specific `addRows` event? As there's no way to listen for this currently.

**Comment by wojciechczerniak** [comment]
Are you sure `valuesUpdated` is not triggered?

![Screenshot 2021-10-18 at 21 25 52](https://user-images.githubusercontent.com/4718998/137794143-3d94ed1a-8c77-4273-ac8a-2e15d87cd1a7.png)

Thing is, if we add empty row, it rarely alters the calculation results, therefore no values are updated in the process
